gateway/src/apicast/http_proxy.lua

```diff
@@ -109,6 +125,12 @@ local function forward_https_request(proxy_uri, uri, skip_https_connect)
         ngx.log(ngx.INFO, "HTTPS Proxy: Request body is bigger than client_body_buffer_size, read the content from path='", temp_file_path, "'")
 
         if temp_file_path then
+          -- Set Content-Length only if none exist
+          if not request.headers["Content-Length"] then
```

Comment:
This check should not be needed. The RFC states that only one of `Transfer-Encoding` & `Content-Length` should be sent and in the case of chunked requests the latter should be omitted or ignored in which case we must always set this.